### PR TITLE
mv: check if --target is a directory

### PR DESCRIPTION
**Pull request by shinhs0506**
should fix mv/diag.sh

### CODE REVIEW

**Comment by sylvestre** [comment]
Still fails it seems:

```
2023-04-26T19:10:55.8539986Z FAIL: tests/mv/diag
2023-04-26T19:10:55.8540067Z ===================
2023-04-26T19:10:55.8540074Z 
2023-04-26T19:10:55.8540249Z --- exp	2023-04-26 19:00:50.588473849 +0000
2023-04-26T19:10:55.8540423Z +++ out	2023-04-26 19:00:50.588473849 +0000
2023-04-26T19:10:55.8540548Z @@ -1,6 +1,17 @@
2023-04-26T19:10:55.8540690Z -mv: missing file operand
2023-04-26T19:10:55.8541018Z -Try 'mv --help' for more information.
2023-04-26T19:10:55.8541226Z -mv: missing destination file operand after 'no-file'
2023-04-26T19:10:55.8541392Z -Try 'mv --help' for more information.
2023-04-26T19:10:55.8541705Z +error: the following required arguments were not provided:
2023-04-26T19:10:55.8541797Z +  <files>...
2023-04-26T19:10:55.8541872Z +
2023-04-26T19:10:55.8542049Z +Usage: mv [OPTION]... [-T] SOURCE DEST
2023-04-26T19:10:55.8542165Z +       mv [OPTION]... SOURCE... DIRECTORY
2023-04-26T19:10:55.8542345Z +       mv [OPTION]... -t DIRECTORY SOURCE...
2023-04-26T19:10:55.8542418Z +
2023-04-26T19:10:55.8542585Z +For more information, try '--help'.
2023-04-26T19:10:55.8543197Z +error: The argument '<files>...' requires at least 2 values, but only 1 was provided
2023-04-26T19:10:55.8543255Z +
2023-04-26T19:10:55.8543427Z +Usage: mv [OPTION]... [-T] SOURCE DEST
2023-04-26T19:10:55.8543538Z +       mv [OPTION]... SOURCE... DIRECTORY
2023-04-26T19:10:55.8543719Z +       mv [OPTION]... -t DIRECTORY SOURCE...
2023-04-26T19:10:55.8543792Z +
2023-04-26T19:10:55.8543956Z +For more information, try '--help'.
2023-04-26T19:10:55.8544108Z  mv: target 'f1': Not a directory
2023-04-26T19:10:55.8544288Z  mv: target directory 'f2': Not a directory
```

**Comment by shinhs0506** [comment]
yep still working on it

**Comment by shinhs0506** [comment]
@sylvestre our program is outputting more error messages than it should (although still helpful) do we try to fix those as well? 
this pr still fixes the last 2 lines of the error messages in the screenshot.
mv-diag

**Comment by cakebaker** [comment]
Can you please add a test to `tests/by-util/test-mv.rs` so we don't regress in the future?

**Comment by cakebaker** [comment]
Thanks for adding the tests, great work!

File: tests/by-util/test_mv.rs [hunk 1]
```diff
@@ -55,6 +55,63 @@ fn test_mv_move_file_into_dir() {
...
+    at.touch(file1);
+    at.touch(file2);
+
+    ucmd.arg("--target")
+        .arg(file1)
+        .arg(file2)
+        .fails()
+        .stderr_is(format!("mv: target directory '{file1}': Not a directory\n"));
+
+    assert!(at.file_exists(format!("{file1}")))
```
Comment (cakebaker) [inline tests/by-util/test_mv.rs:91]:
```suggestion
    assert!(at.file_exists(file1))
```

File: tests/by-util/test_mv.rs [hunk 2]
```diff
@@ -55,6 +55,63 @@ fn test_mv_move_file_into_dir() {
...
+    at.touch(file3);
+
+    ucmd.arg(file1)
+        .arg(file2)
+        .arg(file3)
+        .fails()
+        .stderr_is(format!("mv: target '{file3}': Not a directory\n"));
+
+    assert!(at.file_exists(format!("{file1}")));
+    assert!(at.file_exists(format!("{file2}")));
```
Comment (cakebaker) [inline tests/by-util/test_mv.rs:112]:
```suggestion
    assert!(at.file_exists(file1));
    assert!(at.file_exists(file2));
```

**Comment by sylvestre** [comment]
@shinhs0506 please don't use screenshot. they are terrible for accessibility and search

To answer to your question, as clap is usually giving excellent error, we adjust the GNU test to match our.
Example:
https://github.com/uutils/coreutils/blob/main/util/build-gnu.sh#L223

**Comment by sylvestre** [comment]
`Congrats! The gnu test tests/mv/diag is no longer failing!`
bravo :)

File: util/build-gnu.sh [hunk 3]
```diff
@@ -220,6 +220,9 @@ sed -i -Ez "s/\n([^\n#]*pad-3\.2[^\n]*)\n([^\n]*)\n([^\n]*)/\n# uutils\/numfmt s
 
 # Update the GNU error message to match the one generated by clap
 sed -i -e "s/\$prog: multiple field specifications/error: The argument '--field <FIELDS>' was provided more than once, but cannot be used multiple times\n\nUsage: numfmt [OPTION]... [NUMBER]...\n\n\nFor more information try '--help'/g" tests/misc/numfmt.pl
+sed -i -e "s/Try 'mv --help' for more information/For more information, try '--help'/g" tests/mv/diag.sh
+sed -i -e "s/mv: missing file operand/error: the following required arguments were not provided:\n  <files>...\n\nUsage: mv [OPTION]... [-T] SOURCE DEST\n       mv [OPTION]... SOURCE... DIRECTORY\n       mv [OPTION]... -t DIRECTORY SOURCE...\n/g" tests/mv/diag.sh
+sed -i -e "s/mv: missing destination file operand after 'no-file'/error: The argument '<files>...' requires at least 2 values, but only 1 was provided\n\nUsage: mv [OPTION]... [-T] SOURCE DEST\n       mv [OPTION]... SOURCE... DIRECTORY\n       mv [OPTION]... -t DIRECTORY SOURCE...\n/g" tests/mv/diag.sh
```
Comment (sylvestre) [inline util/build-gnu.sh:225]: Maybe do one call to sed:
```suggestion
sed -i -e "s/Try 'mv --help' for more information/For more information, try '--help'/g" -e "s/mv: missing file operand/error: the following required arguments were not provided:\n  <files>...\n\nUsage: mv [OPTION]... [-T] SOURCE DEST\n       mv [OPTION]... SOURCE... DIRECTORY\n       mv [OPTION]... -t DIRECTORY SOURCE...\n/g" -e "s/mv: missing destination file operand after 'no-file'/error: The argument '<files>...' requires at least 2 values, but only 1 was provided\n\nUsage: mv [OPTION]... [-T] SOURCE DEST\n       mv [OPTION]... SOURCE... DIRECTORY\n       mv [OPTION]... -t DIRECTORY SOURCE...\n/g" tests/mv/diag.sh
```

(it is ugly but well... :)

**Pull request merged**